lib/freedom_patches/pluck_first.rb

```diff
@@ -2,18 +2,16 @@
 
 class ActiveRecord::Relation
   def pluck_first(*attributes)
-    limit(1).pluck(*attributes).first
+    Discourse.deprecate("`#pluck_first` is deprecated, use `#pick` instead.")
+    pick(*attributes)
   end
 
   def pluck_first!(*attributes)
-    items = limit(1).pluck(*attributes)
+    Discourse.deprecate("`#pluck_first!` is deprecated without replacement.")
+    items = pick(*attributes)
 
-    raise_record_not_found_exception! if items.empty?
+    raise_record_not_found_exception! if items.blank?
 
-    items.first
+    items
   end
 end
-
-module ActiveRecord::Querying
```

Comment:
We can’t remove this yet, as it’s still used by some plugins (see the failing plugin jobs on the CI) :wink: